lib/rules-fr.js

```diff
@@ -0,0 +1,105 @@
+// A set of predefined rules that are especially applicable to French strings.
+
+const en = require('./rules-en');
+const { RemoveRule, FunctionRule } = require('./rules');
+const normalize = require('./normalize');
+
+const COMMON_PUNCTUATION_STR = en.COMMON_PUNCTUATION_STR + `«»`;
+const COMMON_SYMBOLS_STR = en.COMMON_SYMBOLS_STR + `€`;
+const COMMON_PUNCTUATION_AND_SYMBOLS_STR =
+  COMMON_PUNCTUATION_STR + COMMON_SYMBOLS_STR;
+const COMMON_PUNCTUATION_AND_SYMBOLS = new RemoveRule(
+  COMMON_PUNCTUATION_AND_SYMBOLS_STR,
+  {
+    name: 'common punctuation and symbols',
+  }
+);
+
+const replaceLigatures = str => {
+  return str
+    .replace(/æ/g, 'ae')
+    .replace(/Æ/g, 'AE')
+    .replace(/œ/g, 'oe')
+    .replace(/Œ/g, 'OE');
+};
+
+const LIGATURES = new FunctionRule(
+  (s1, s2) => {
+    return [replaceLigatures(s1), replaceLigatures(s2)];
+  },
+  { name: 'ligatures replaced' }
+);
```

Comment:
Since the replace rule only operates on one character at a time, I made this a FunctionRule so that both ligature candidates could be specified. Note that all FunctionRules are run _after_ the other rules.